### PR TITLE
Np 186 seperate cura resources

### PR DESCRIPTION
**Pull request by jellespijker**
# Description

Splits up the creation of the Conan package for printer, profile, resources. This will allow for multiple consumers to these shared resources. It will also lighten the the Cura conan packages, since it now doesn't need to package the meshes and definitions when there are only code changes. The version number is reused from the conandata.yml defined in the Cura root.

Downside is that we also need to trigger the cura_resources for release packages, in the cura_workflows.

See the following related PR's:
- https://github.com/Ultimaker/CuraSettings/pull/61
- https://github.com/Ultimaker/cura-workflows/pull/17
- https://github.com/Ultimaker/fdm_materials/pull/293
- https://github.com/Ultimaker/CuraEngine/pull/2077

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Printer definition file(s)
- [ ] Translations

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Operating System:

# Checklist:

- [X] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta) and [Cura QML best practices](https://github.com/Ultimaker/Cura/wiki/QML-Best-Practices)
- [X] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/CONTRIBUTING.md) 
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have uploaded any files required to test this change

### CODE REVIEW

File: resources/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.58.0 <2.0.0"
+
+
+class CuraResource(ConanFile):
+    name = "cura_resources"
+    license = ""
+    author = "UltiMaker"
+    url = "https://github.com/Ultimaker/cura-private-data"
```
Comment (casperlamboo) [inline resources/conanfile.py:15]: is this url right?

File: resources/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    description = "Cura Resources"
+    topics = ("conan", "cura")
+    exports = "LICENSE*"
```
Comment (casperlamboo) [inline resources/conanfile.py:18]: the license is located one directory up, is that an issue? Also this conan package is just data, as I understand it data is not copy-rightable. Do we even need a license?

File: resources/conanfile.py (hunk 1 above)
Comment (jellespijker) [reply resources/conanfile.py:15]: nope, will change

File: resources/conanfile.py (hunk 2 above)
Comment (jellespijker) [reply resources/conanfile.py:18]: can't hurt to have the license here, will change

File: conandata.yml [hunk 3]
```diff
@@ -50,10 +51,30 @@ pyinstaller:
...
+            src: "res/meshes"
+            dst: "share/cura/resources/meshes"
+        cura_resources_quality:
+          package: "cura_resources"
+          src: "res/quality"
+          dst: "share/cura/resources/quality"
+        cura_resources_variants:
+          package: "cura_resources"
+          src: "res/variants"
+          dst: "share/cura/resources/variants"
```
Comment (wawanbreton) [inline conandata.yml:77]: No way to define those all in a single shot ?

File: resources/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        copy(self, pattern="*", src=os.path.join(self.recipe_folder, "images"),
+             dst=os.path.join(self.export_sources_folder, "images"))
+        copy(self, pattern="*", src=os.path.join(self.recipe_folder, "intent"),
+             dst=os.path.join(self.export_sources_folder, "intent"))
+        copy(self, pattern="*", src=os.path.join(self.recipe_folder, "meshes"),
+             dst=os.path.join(self.export_sources_folder, "meshes"))
+        copy(self, pattern="*", src=os.path.join(self.recipe_folder, "quality"),
+             dst=os.path.join(self.export_sources_folder, "quality"))
+        copy(self, pattern="*", src=os.path.join(self.recipe_folder, "variants"),
+             dst=os.path.join(self.export_sources_folder, "variants"))
```
Comment (wawanbreton) [inline resources/conanfile.py:45]: I would suggest to define this list of subfolders globally and use loops to process them. As far as I can see, they are used in 3 different places so this is very error-prone.

File: resources/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    def validate(self):
+        if Version(self.version) <= Version("4"):
+            raise ConanInvalidConfiguration("Only versions 5+ are support")
+
+    def layout(self):
+        self.cpp.source.resdirs = ["definitions", "extruders", "images", "intent", "meshes", "quality", "variants"]
+        self.cpp.package.resdirs = ["res/definitions", "res/extruders", "res/images", "res/intent", "res/meshes", "res/quality", "res/variants"]
```
Comment (wawanbreton) [inline resources/conanfile.py:53]: :arrow_up:

**Review by wawanbreton — COMMENTED** [review]
Some minor remarks, but this all makes sense. I'm happy that you didn't split the resources to an other repository :blush:

File: conandata.yml (hunk 3 above)
Comment (jellespijker) [reply conandata.yml:77]: Should be possible

**Pull request merged**